### PR TITLE
MODGOBI-86-Separate ISBN qualifier

### PR DESCRIPTION
**Pull request by KVupp**
## Purpose
https://issues.folio.org/browse/MODGOBI-86

## Approach

- Separate the ISBN qualifier from separate field
- If the qualifier is in the same field as ProductId, only assign the ID to the ProductID field and not the qualifier

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java [hunk 1]
```diff
@@ -340,7 +340,22 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
...
   public CompletableFuture<String> lookupMock(String data) {
     logger.info("Mocking the data lookup for: " + data);
-    return CompletableFuture.completedFuture(UUID.randomUUID().toString());
+    return CompletableFuture.completedFuture(UUID.randomUUID()
+      .toString());
+  }
+
+  public CompletableFuture<String> separateISBNQualifier(String data) {
+    if (data.split("\\s+").length > 1) {
+      return completedFuture(data.split("\\s+")[1]);
```
Comment (pautau-s) [inline src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: If the classifier consists of more than one word, the array will contain more than two strings. For example: 
given
```
<subfield code="a"> 9781410352224 (some qualifier)</subfield>
```
result
```
getProductId - 9781410352224 
qualifier - (some
```
expected result
```
getProductId - 9781410352224 
qualifier - (some qualifier)
```

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java [hunk 2]
```diff
@@ -340,7 +340,22 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
...
+    }
+    return completedFuture(null);
+  }
+
+  public CompletableFuture<String>  truncateISBNQualifier(String data) {
+    if(data.split("\\s+").length > 1) {
+      return completedFuture(data.split("\\s+")[0]);
```
Comment (pautau-s) [inline src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: If the first character is a space, then after splitting the string, the `productId` field will get an empty string and the `qualifier` will get ISBN number.

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java [hunk 3]
```diff
@@ -817,6 +821,24 @@ public final void testPostGobiOrdersGetExistingPOLineNumberFails() throws Except
...
+
+    final Async asyncLocal = context.async();
+
+    final String body = getMockData(PO_LISTED_ELECTRONIC_MONOGRAPH_QUALIFIER_PATH);
+    postOrderSuccess(body);
+
+    Map<String, List<JsonObject>> postOrder = MockServer.serverRqRs.column(HttpMethod.POST);
+    CompositePurchaseOrder compPO = postOrder.get(PURCHASE_ORDER).get(0).mapTo(CompositePurchaseOrder.class);
+    assertNotNull(compPO.getCompositePoLines().get(0).getDetails().getProductIds().get(0).getQualifier());
+    assertEquals("9781410352224",compPO.getCompositePoLines().get(0).getDetails().getProductIds().get(0).getProductId());
```
Comment (pautau-s) [inline src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: It seems that the value of the qualifier is not checked anywhere.

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java (hunk 1 above)
Comment (KVupp) [reply src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: Done, added length to split function

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java (hunk 2 above)
Comment (KVupp) [reply src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: done

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java (hunk 3 above)
Comment (KVupp) [reply src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: added test cases covering various scenarios

**Pull request merged**